plugins/window-state/src/lib.rs

```diff
@@ -106,10 +108,12 @@ impl<R: Runtime> WindowExt for Window<R> {
             // otherwise, let the OS decide where to place the window
             for m in self.available_monitors()? {
                 if m.name().map(ToString::to_string).unwrap_or_default() == state.monitor {
-                    self.set_position(PhysicalPosition {
-                        x: state.x,
-                        y: state.y,
-                    })?;
+                    if *m.position() == state.monitor_position && *m.size() == state.monitor_size {
```

Comment:
This may need a refactor for readability soon.
Essentially we're invalidating the cache, based on a more detailed *fingerprint* of what the monitor setup was when we saved the state. Vs when we're restoring the state.

Importantly we're only ever checking if it's **exactly equal** to the aspects we fingerprinted.

Reply:
and thats okay, the problem is that for some reason Window lets you go out of the physical screen.
I mainly use linux too but I dont have a dual monitor setup on it